src/types/acls.rs

```diff
@@ -0,0 +1,350 @@
+                    Response::Acl(a) => Ok(MapOrNot::Map(InnerAcl {
+                        mailbox: a.mailbox,
+                        acls: a
+                            .acls
+                            .into_iter()
+                            .map(|e| AclEntry {
+                                identifier: e.identifier,
+                                rights: e.rights.into(),
+                            })
+                            .collect(),
+                    })),
+                    resp => Ok(MapOrNot::Not(resp)),
+                })
+            },
+        }
+        .try_build()
+    }
+
+    /// Return the mailbox the ACL entries belong to
+    pub fn mailbox(&self) -> &str {
+        &*self.borrow_acl().mailbox
+    }
+
+    /// Returns a list of identifier/rights pairs for the mailbox
+    pub fn acls(&self) -> &[AclEntry<'_>] {
+        &*self.borrow_acl().acls
+    }
+}
+
+/// Inner struct to manage storing the references for ouroboros
+#[derive(Debug, Eq, PartialEq)]
+pub(crate) struct InnerAcl<'a> {
+    /// The mailbox the ACL Entries belong to
+    pub(crate) mailbox: Cow<'a, str>,
+    /// The list of identifier/rights pairs for the mailbox
+    pub(crate) acls: Vec<AclEntry<'a>>,
+}
+
+/// From [section 3.6 of RFC 4313](https://datatracker.ietf.org/doc/html/rfc4314#section-3.6).
+///
+/// The list of identifiers and rights for the [`Acl`] response
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct AclEntry<'a> {
+    /// The user identifier the rights are for
+    pub identifier: Cow<'a, str>,
+    /// the rights for the provided identifier
+    pub rights: AclRights,
+}
+
+/// From [section 3.7 of RFC 4313](https://datatracker.ietf.org/doc/html/rfc4314#section-3.7).
+///
+/// The LISTRIGHTS response from the [`Session::list_rights`] IMAP command
+#[self_referencing]
+pub struct ListRights {
+    data: Vec<u8>,
+    #[borrows(data)]
+    #[covariant]
+    pub(crate) rights: InnerListRights<'this>,
+}
+
+impl ListRights {
+    /// Parse the given input into a [`ListRights`] response.
+    pub fn parse(
+        owned: Vec<u8>,
+        unsolicited: &mut mpsc::Sender<UnsolicitedResponse>,
+    ) -> Result<Self, Error> {
+        ListRightsTryBuilder {
+            data: owned,
+            rights_builder: |input| {
+                // There should only be ONE single LISTRIGHTS response
+                parse_until_done(input, unsolicited, |response| match response {
+                    Response::ListRights(a) => Ok(MapOrNot::Map(InnerListRights {
+                        mailbox: a.mailbox,
+                        identifier: a.identifier,
+                        required: a.required.into(),
+                        optional: a.optional.into(),
+                    })),
+                    resp => Ok(MapOrNot::Not(resp)),
+                })
+            },
+        }
+        .try_build()
+    }
+
+    /// Returns the mailbox for the rights
+    pub fn mailbox(&self) -> &str {
```

Comment:
Given the change we made around `Quota`, how do you feel about having just one accessor here that gives out `&InnerListRights` (and giving it a friendlier name)?

Reply:
If we went that route then it might make more sense to simply name the structs returned by the Session methods to be *Response (ListRightsResponse, MyRightsResponse) etc.. And then the "inner" structs will be renamed to ListRights, MyRights, etc..

Let me know what you think.

Reply:
@jonhoo I'll push a fixup commit that performs this change (*Response structs) so you can see it and then we can decide to keep or go another direction

Reply:
That seems reasonable to me!